### ISSUE TITLE
[Bug]: data race in UT TestCommitS3Blocks

### ISSUE THREAD

**Issue by volgariver6** [matrixorigin/matrixone#9868]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

data race happens in UT case TestCommitS3Blocks.

https://github.com/matrixorigin/matrixone/actions/runs/5183718751/jobs/9341924211?pr=9866

<img width="894" alt="image" src="https://github.com/matrixorigin/matrixone/assets/18366608/c9834612-be39-436b-847e-445e4046be53">

### Expected Behavior

no data race.

**Comment by volgariver6** [comment]
@jiangxinmeng1 please check this issue.

**Comment by jiangxinmeng1** [comment]
fixed in #9880
